anta/reporter/md_reporter.py

```diff
@@ -0,0 +1,332 @@
+    to generate and write content to the provided markdown file.
+    """
+
+    ONLY_FAILED_TESTS: ClassVar[bool] = False
+
+    def __init__(self, mdfile: TextIOWrapper, results: ResultManager) -> None:
+        """Initialize the MDReportBase with an open markdown file object to write to and a ResultManager instance.
+
+        Parameters
+        ----------
+            mdfile: An open file object to write the markdown data into.
+            results: The ResultsManager instance containing all test results.
+        """
+        self.mdfile = mdfile
+        self.results = results
+
+    @abstractmethod
+    def generate_section(self) -> None:
+        """Abstract method to generate a specific section of the markdown report.
+
+        Must be implemented by subclasses.
+        """
+        msg = "Must be implemented by subclasses"
+        raise NotImplementedError(msg)
+
+    def generate_rows(self) -> Generator[str, None, None]:
+        """Generate the rows of a markdown table for a specific report section.
+
+        Subclasses can implement this method to generate the content of the table rows.
+        """
+        msg = "Subclasses should implement this method"
+        raise NotImplementedError(msg)
+
+    def generate_heading_name(self) -> str:
+        """Generate a formatted heading name based on the class name.
+
+        Returns
+        -------
+            str: Formatted header name.
+
+        Example:
+        -------
+            - `ANTAReport` will become ANTA Report.
+            - `TestResultsSummary` will become Test Results Summary.
+        """
+        class_name = self.__class__.__name__
+
+        # Split the class name into words, keeping acronyms together
+        words = re.findall(r"[A-Z]?[a-z]+|[A-Z]+(?=[A-Z][a-z]|\d|\W|$)|\d+", class_name)
+
+        # Capitalize each word, but keep acronyms in all caps
+        formatted_words = [word if word.isupper() else word.capitalize() for word in words]
+
+        return " ".join(formatted_words)
+
+    def write_table(self, table_heading: list[str], *, last_table: bool = False) -> None:
+        """Write a markdown table with a table heading and multiple rows to the markdown file.
+
+        Parameters
+        ----------
+            table_heading: List of strings to join for the table heading.
+            last_table: Flag to determine if it's the last table of the markdown file to avoid unnecessary new line. Defaults to False.
+        """
+        self.mdfile.write("\n".join(table_heading) + "\n")
+        for row in self.generate_rows():
+            self.mdfile.write(row)
+        if not last_table:
+            self.mdfile.write("\n")
+
+    def write_heading(self, heading_level: int) -> None:
+        """Write a markdown heading to the markdown file.
+
+        The heading name used is the class name.
+
+        Parameters
+        ----------
+            heading_level: The level of the heading (1-6).
+
+        Example:
```

Comment:
```suggestion
        Example
```

Reply:
Fixed in 8918cd0822dee86db2c1574d30732c0418bfefff